andela/suyabay
# [#120180435] create endpoint that retrieves comments in an episode within a date range

#### What does this PR do?
- Leverage on the previous endpoint (`/v1/episodes/:name/comments`) to retrieve all the comments available in a particular episode within a date range and a particular limit passed by the user as queries.
- Write tests that test the headcase (get comments if the query is passed).
#### Description of Task to be completed?
-  Leverage on the previous route
-  Create additional two methods that return various data and messages for different implementation and head cases leveraging on the previously written method.
-  Use `fractal package (Transformer)`  to transform the comments result in each query. This is performed by one of the new method created
-  Write a test  that asserts that the right data and right status code is presented when the query is passed.
#### How should this be manually tested?
- We need POSTMAN for this to be manually tested, POSTMAN is an extension plugin on google chrome.
-  To retrieve all comments under a specified date,  send a `GET` request with this endpoint  `suyabay.app/api/v1/episodes/Gregorio Von/comments?fromDate=2016-05-10&toDate=2016-05-18`. If the {episode-name} is valid, you should see a JSON response with the data requested from the recent comment to the oldest commence, a status code of `200`. If you add a limit, to the query like so:       `suyabay.app/api/v1/episodes/Gregorio Von/comments?fromDate=2016-05-10&toDate=2016-05-18&limi=2`. If the {episode-name} is valid, you should see a JSON response with the data requested from the recent comment to the oldest commence with the limit of 2, status code of `200`.
#### Any background context you want to provide?

None
#### What are the relevant pivotal tracker stories?
- #120180435
#### Screenshots (if appropriate)

![screen shot 2016-05-25 at 11 01 40 am](https://cloud.githubusercontent.com/assets/14159420/15537174/c6c2168a-226c-11e6-8a84-8f28e0865ef0.png)

![screen shot 2016-05-25 at 11 02 03 am](https://cloud.githubusercontent.com/assets/14159420/15537175/c6c614e2-226c-11e6-9f15-1287aea8ea81.png)
#### Questions:

[@unicodeveloper](https://github.com/unicodeveloper)
[@andela-oadebayo](https://github.com/andela-oadebayo)
